onnxscript/rewriter/pattern.py

```diff
@@ -1119,6 +1119,58 @@ def replace_pattern(new_pattern):
         return [replace_pattern(p) for p in self._target_pattern.commute()]
 
 
+class RewriteRuleAsClass:
+    """Defines a class grouping method pattern, rewrite, check.
+    This class is then given to function :func:`make_rewrite_rule_from_class`
+    to define a new rule.
+    """
+
+    @classmethod
+    def pattern(cls, op, *_) -> Any:
+        raise NotImplementedError("Method 'pattern' must be overwritten.")
+
+    @classmethod
+    def rewrite(cls, op, *_) -> Any:
+        raise NotImplementedError("Method 'rewrite' must be overwritten.")
+
+    @classmethod
+    def check(cls, context, *_) -> bool:
+        return True
+
+
+def make_rewrite_rule_from_class(rule_class: type | RewriteRuleAsClass) -> RewriteRule:
```

Comment:
Is the plan to support both forms? Or, switch everything to the Class form?

Reply:
Support both forms. The one I propose is just to make it easier to read if multiple pattern are implemented in the same file.